src/pages/SearchPage/index.jsx

```diff
@@ -0,0 +1,192 @@
+// third-party libraries
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+
+// scss
+import './SearchPage.scss';
+
+// components
+import Spinner from 'components/Spinner';
+import Card from 'components/Card';
+import Search from 'components/Search';
+
+// utils
+import connect from 'utils/connect';
+import { searchArticlesRequest } from 'modules/search';
+
+/**
+ * Component for Search page
+ * @summary Handle article search
+ */ export class SearchPage extends Component {
+  /**
+   * @name Vote propTypes
+   * @type {propTypes}
+   *
+   * @param {Object} props - React PropTypes
+   *
+   * @property {string} term - search query
+   * @property {string} query - user input
+   * @property {boolean} searched - to if the user is searched
+   * @property {Function} formSubmit - handle term
+   * @property {Function} handleInputChange - handle input change
+   * @property {Function} handleTermChange - handle term
+   * @property {Function} handleSearch - process the search
+   * @property {Function} handleIncomingSearch - handle the search on the global header
+   * @property {Function} renderList - render searched item
+   *
+   */
+  static propTypes = {
+    search: PropTypes.instanceOf(Object).isRequired,
+    searchArticlesRequest: PropTypes.func.isRequired,
+    history: PropTypes.instanceOf(Object).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    const {
+      history: {
+        location: { search },
+      },
+    } = this.props;
+    const query = search.split('=')[1];
+    this.state = {
+      term: 'tag',
+      query,
+      searched: false,
+    };
+  }
+
+  componentDidMount() {
+    const { query } = this.state;
+    if (query) {
+      this.handleIncomingSearch();
+    }
+  }
+
+  submitForm = (term) => {
+    const { query } = this.state;
+    const { searchArticlesRequest: searchArticles } = this.props;
+    searchArticles({ term, query });
+  };
+
+  handleInputChange = ({ target: { value } }) => {
+    this.setState({ query: value });
+  };
+
+  handleTermChange = ({ target: { value } }) => {
+    this.setState(() => ({
+      term: value,
+    }));
+
+    this.submitForm(value);
+  };
+
+  handleSearch = ({ keyCode }) => {
+    const { term } = this.state;
+    if (keyCode === 13) {
+      this.setState(() => ({
+        searched: true,
+      }));
+
+      return this.submitForm(term);
+    }
+    return null;
+  };
+
+  handleIncomingSearch = () => {
+    const { term } = this.state;
+    this.setState(() => ({
+      searched: true,
+    }));
+
+    this.submitForm(term);
+  };
+
+  renderList = () => {
+    const {
+      search: { data }
+    } = this.props;
+
+    return (data)
```

Comment:
We also don't need the &&. `data` is an array by default so this would still be mapped even if the array is empty.

<img width="257" alt="Screenshot 2019-09-11 at 10 03 54 AM" src="https://user-images.githubusercontent.com/22734663/64683626-880b0400-d47b-11e9-8fbd-3cd729fad45c.png">

So either fix your logic or remove the &&